### PR TITLE
[fix] vec * mat in matmul in onnx converter

### PR DESCRIPTION
**Pull request by ganler**
Should fix https://github.com/apache/tvm/issues/10651 for ONNX converter.

cc: @masahi

### CODE REVIEW

File: python/tvm/relay/frontend/onnx.py [hunk 1]
```diff
@@ -324,6 +324,9 @@ def flatten_to_nd(x, x_shape, nd=3):
             0,
         )
         return _op.reshape(output, fold_constant(final_shape))
+    elif a_rank == 1:
+        return _op.multiply(inputs[0], inputs[1])
```
Comment (masahi) [inline python/tvm/relay/frontend/onnx.py:329]: So `op.multiply` with shapes `(3), (3, 1)` does vec * mat?

Comment (ganler) [reply python/tvm/relay/frontend/onnx.py:329]: Yes, this is also valid in ONNX standard. Both PyTorch and ONNXRuntime can pass it.

Comment (ganler) [reply python/tvm/relay/frontend/onnx.py:329]: In [ONNX](https://github.com/onnx/onnx/blob/main/docs/Changelog.md#MatMul-13)'s spec, it says it simply follows numpy (PyTorch also largely follows numpy).

In NumPy's [doc](https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.matmul.html), there is one case: 

> If the first argument is 1-D, it is promoted to a matrix by prepending a 1 to its dimensions. After matrix multiplication the prepended 1 is removed.

**Review by masahi — APPROVED** [review]

File: python/tvm/relay/frontend/onnx.py (hunk 1 above)
Comment (ganler) [reply python/tvm/relay/frontend/onnx.py]: @masahi Oh sorry, I double-checked your comment and my previous fix of `op. multiply` is actually incorrect as its output shape is still `[3, 1]` not `[1]` (I passed the compile but the output shape is incorrect). Just fixed that with `_op.nn.matmul`.

I also noticed that in TVM, the behaviour of matmul is more explicit than numpy that `vec` as `tensor_a` is unacceptable. Hence, I need to do unsqueeze and squeeze tricks here...

**Comment by ganler** [comment]
@masahi CI finally passed! I think we can also close the issue after the merge. :-)

**Pull request merged**